### ISSUE TITLE
UI bug: the Source/Destination CIDR are set to empty not 0.0.0.0/0

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#10002]
The issue happens when create a firewall rule for source nat, static nat, lb or pf on UI.

for example for port forwarding

- specify the public/private port 22 , the rule is created with source cidr 0.0.0.0/0, which is fine.
- specify the public/private port 23 , input some value as source cidr then clean the field, the rule is created with empty source cidr.

![image](https://github.com/user-attachments/assets/e991d2a4-d82d-435f-90d1-a9389cc089ec)

in the VR, there is no iptables rules for port 23, which is wrong.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.20, should exist in 4.19 as well
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by iishitahere** [comment]
Hi @weizhouapache, I’d like to work on this issue. From my understanding, the Source/Destination CIDR fields should default to 0.0.0.0/0 instead of being empty. Could you confirm if this is the expected behavior? If assigned, I’ll get started right away. Looking forward to contributing!

**Comment by DaanHoogland** [comment]
> Hi @weizhouapache, I’d like to work on this issue. From my understanding, the Source/Destination CIDR fields should default to 0.0.0.0/0 instead of being empty. Could you confirm if this is the expected behavior? If assigned, I’ll get started right away. Looking forward to contributing!

@iishitahere , that is correct. When first editting and thanclearing the field, the default of 0.0.0.0/0 should be used again. This can be a UI fix, but as I understand the problem it can also be solved in the API. What do you think @weizhouapache ?

**Comment by weizhouapache** [comment]
> > Hi @weizhouapache, I’d like to work on this issue. From my understanding, the Source/Destination CIDR fields should default to 0.0.0.0/0 instead of being empty. Could you confirm if this is the expected behavior? If assigned, I’ll get started right away. Looking forward to contributing!
> 
> @iishitahere , that is correct. When first editting and thanclearing the field, the default of 0.0.0.0/0 should be used again. This can be a UI fix, but as I understand the problem it can also be solved in the API. What do you think @weizhouapache ?

@DaanHoogland @iishitahere 
from what I know, most APIs with source/destination CIDR are able to process an empty list.
so it looks like a UI issue, UI passes a list with empty string as api parameter. it should not pass the value if the cidr is an empty string or only contains spaces.

**Comment by iishitahere** [comment]
> > > i @weizhouapache, I’d like to work on this issue. From my understanding, the Source/Destination CIDR fields should default to 0.0.0.0/0 instead of being empty. Could you confirm if this is the expected behavior? If assigned, I’ll get started right away. Looking forward to contributing!
> > 
> > 
> > @iishitahere , that is correct. When first editting and thanclearing the field, the default of 0.0.0.0/0 should be used again. This can be a UI fix, but as I understand the problem it can also be solved in the API. What do you think @weizhouapache ?
> 
> @DaanHoogland @iishitahere from what I know, most APIs with source/destination CIDR are able to process an empty list. so it looks like a UI issue, UI passes a list with empty string as api parameter. it should not pass the value if the cidr is an empty string or only contains spaces.

Thank you, @weizhouapache  and @DaanHoogland , for the clarification! Based on your input, it seems that addressing this as a UI fix would be the right approach. I’ll focus on ensuring that the UI does not pass an empty string or spaces as API parameters for the CIDR field and defaults to 0.0.0.0/0 when the field is cleared.

I’ll start working on this and keep you updated on my progress. Please let me know if there are any additional details or specific areas you'd like me to focus on. Thanks again for assigning me to this issue!

**Comment by DaanHoogland** [comment]
fixed in #10208
